### PR TITLE
docs: Add missing angular documentation

### PR DESCRIPTION
**Pull request by wlee221**
*Issue #, if available:*

*Description of changes:* Adds missing angular markdowns. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/src/pages/ui/components/authenticator/headless.angular.mdx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+The `Authenticator`'s UI is powered by a state machine. You can use the `StateMachineService`composable
+to get access to it and customizing behavior or your own UI:
+
+_app.component.ts_
+
+```ts
+import { StateMachineService } from '@aws-amplify/ui-angular';
+
+export class AppComponent {
+  constructor(private stateMachine: StateMachineService) {}
```
Comment (wlee221) [inline docs/src/pages/ui/components/authenticator/headless.angular.mdx:10]: This service will likely be changed with a more publicly facing API, e.g. `AuthenticatorService`.

File: examples/angular/src/app/app.module.ts [hunk 2]
```diff
@@ -5,6 +5,7 @@ import { SignInWithEmailComponent } from 'src/pages/ui/components/authenticator/
 
 import { AppRoutingModule } from './app-routing.module';
 import { AppComponent } from './app.component';
+// TODO: Rename UiAngularModule to AmplifyUiAngularModule
```
Comment (wlee221) [inline examples/angular/src/app/app.module.ts:8]: Noticed my angular module is named too generally. Will add a PR by this week.

**Review by ErikCH — CHANGES_REQUESTED** [review]
Just 2 quick nit picks. 

The links in the Angular documentation for source directs to Githubs 404 page. The Demo and Test work however. I believe this is because the examples app uses a different directory structure.

image

And the first example for getting started, doesn't actually show wrapping the template with `amplify-authenticator`. Instead it shows how to install it in the app module. Maybe it should show both?

**Comment by wlee221** [comment]
Re: @ErikCH,

> The links in the Angular documentation for source directs to Githubs 404 page.

Fixed in https://github.com/aws-amplify/amplify-ui/pull/322/commits/d5b45d04783f3b5f82372120dc4727c0c1d54fc8.

> And the first example for getting started, doesn't actually show wrapping the template with amplify-authenticator. 

Can't believe I missed this 🤦  Thanks for the catch! Done in https://github.com/aws-amplify/amplify-ui/pull/322/commits/63788dfbf4effb9d49f649aa6f6aa8e981e85f25.

**Review by ErikCH — APPROVED** [review]
Looks good!

File: docs/src/pages/ui/components/authenticator/getting-started.angular.mdx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+```css
+@import '~@aws-amplify/ui-angular/theme.css';
+```
+
+3. Wrap your template with `amplify-authenticator` inside `app.component.html`
+
+```html
+<amplify-authenticator>
+  <ng-template amplifySlot="authenticated" let-user="user">
```
Comment (wlee221) [inline docs/src/pages/ui/components/authenticator/getting-started.angular.mdx:34]:
```suggestion
  <ng-template
    amplifyOverride="authenticated"
    let-user="user"
    let-signOut="signOut"
  >
```

**Pull request merged**